akd_core/src/ecvrf/ecvrf_impl.rs

```diff
@@ -231,18 +233,41 @@ impl VRFPublicKey {
                 .chain(alpha)
                 .chain([counter, ZERO])
                 .finalize();
-            result.copy_from_slice(&hash[..32]);
-            let wrapped_point = CompressedEdwardsY::from_slice(&result)
-                .expect("Result hash should have a length of 32, but it does not")
-                .decompress();
+            hash_result.copy_from_slice(&hash[..32]);
+            let wrapped_point = interpret_hash_value_as_a_point(hash_result);
             counter += 1;
             if let Some(wp) = wrapped_point {
-                return wp.mul_by_cofactor();
+                let result = wp.mul_by_cofactor();
+
+                // Ensure that what we are returning is not the identity point
+                if !result.is_identity() {
+                    return result;
+                }
             }
         }
     }
 }
 
+/// As defined in [Section 5.1.3 of RFC8032](https://www.rfc-editor.org/rfc/rfc8032#section-5.1.3)
+///
+/// Will return Some(point) if the hash value can be interpreted as a point, and None otherwise.
+fn interpret_hash_value_as_a_point(hash: [u8; 32]) -> Option<EdwardsPoint> {
+    // If the input bytes are such that bytes 1 to 30 have value 255, byte 31 has value 255 or 127,
+    // and byte 0 has value 256 - i for value i in the (1, 3, 4, 5, 9, 10, 13, 14, 15, 16) list, then
+    // the encoding is invalid.
+    let is_invalid = hash[1..=30].iter().all(|b| *b == 255)
+        && (hash[31] == 255 || hash[31] == 127)
+        && [1, 3, 4, 5, 9, 10, 13, 14, 15, 16]
+            .iter()
+            .any(|i| hash[0] == (256 - i) as u8);
+    if is_invalid {
```

Comment:
nit: Given it's the last statement of the function, not sure the early return makes it easier to read? Feel free to ignore this one, might just be me

```rust
if is_invalid {
  None
} else {
  CompressedEdwardsY::from_slice(&hash).expect("...").decompress()
}
```

Reply:
Hmm, yeah, style-wise I prefer not typing out the "else" and instead typing out the "return", I guess